### PR TITLE
chore: put gen header to CMAKE_CURRENT_BINARY_DIR

### PR DESCRIPTION
**Pull request by Decodetalkers**
Log:

### CODE REVIEW

**Comment by Decodetalkers** [comment]
@18202781743

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -89,7 +89,7 @@ target_link_directories(${LIB_NAME} INTERFACE
 # gen dtkdeclarative_config.h
 include(DtkBuildConfig)
 set(options MSG) # print MODULE_NAME D_HEADERS DEST_DIR
-gen_dtk_config_header(MODULE_NAME ${LIB_NAME} HEADERS ${D_HEADERS} DEST_DIR ${CMAKE_CURRENT_LIST_DIR}/src OUTPUT_VARIABLE CONFIG_PATH)
+gen_dtk_config_header(MODULE_NAME ${LIB_NAME} HEADERS ${D_HEADERS} DEST_DIR ${CMAKE_CURRENT_BINARY_DIR} OUTPUT_VARIABLE CONFIG_PATH)
```
Comment (18202781743) [inline src/CMakeLists.txt:92]: 还有qt6/src/CMakeLists.txt，

Comment (Decodetalkers) [reply src/CMakeLists.txt:92]: 好了

**Review by 18202781743 — APPROVED** [review]

**Pull request merged**